### PR TITLE
Refactor `special_step_sounds` out of existence, Human steps now depend on leg type, Digitigrade legs use claw footstep sfx

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

1. Deletes `special_step_sounds`. Unused anyways. 
2. Human footstep type is now determined by its legs. 
    - This means a human with two different legs, with two different footstep types, now alternate between footstep type every other step. 
3. Digitigrade legs now use claw footstep sfx

## Why It's Good For The Game

Less random species vars, more dynamic human code, and a little bit more immersion 

## Changelog

:cl: Melbert
refactor: Refactored footsteps for humans. Human footstep sound effects are now determined by your leg type. Report any oddities.
qol: Digitigrade legs now play claw footstep SFX. "plat plat" is dead, long live "tap tap".
/:cl:

### CODE REVIEW

File: code/datums/elements/footstep.dm [hunk 1]
```diff
@@ -74,31 +74,41 @@
 			playsound(turf, 'sound/effects/footstep/crawl1.ogg', 15 * volume, falloff_distance = 1, vary = sound_vary)
 		return
 
-	if(iscarbon(source))
-		var/mob/living/carbon/carbon_source = source
-		if(!carbon_source.get_bodypart(BODY_ZONE_L_LEG) && !carbon_source.get_bodypart(BODY_ZONE_R_LEG))
-			return
-		if(carbon_source.move_intent == MOVE_INTENT_WALK)
-			return// stealth
+	if(iscarbon(source) &&& source.move_intent == MOVE_INTENT_WALK)
```
Comment (Sadboysuss) [inline code/datums/elements/footstep.dm:77]: &&&

**Comment by Admiral-Gigaton** [comment]
*thump click thump click thump click* "oh hey its john, you can tell by the alternating lizard and boot sounds" "fuck you tim"

**Comment by Dawnseer** [comment]
Peg leg buff if ever I saw one. And perhaps a small lizard buff. But primarily peg leg buff!

**Comment by MrMelbert** [comment]
Peg leg 100% needs a unique footstep sound

File: code/datums/elements/footstep.dm [hunk 2]
```diff
@@ -74,31 +74,41 @@
...
-		if(carbon_source.move_intent == MOVE_INTENT_WALK)
-			return// stealth
+	if(iscarbon(source) && source.move_intent == MOVE_INTENT_WALK)
+		return // stealth
+
 	steps_for_living[source] += 1
 	var/steps = steps_for_living[source]
 
-	if(steps >= 6)
+	if(steps >= 8)
```
Comment (Ghommie) [inline code/datums/elements/footstep.dm:83]: Considering that while floating, the sound is only played every six steps, I think this should be 24 (common multiple of both) to allow both left foot and right foot sounds to play evenly.

```suggestion
	if(steps >= 24)
```
Feel free to update the comment below however you want.

File: code/datums/elements/footstep.dm [hunk 3]
```diff
@@ -125,44 +135,68 @@
...
 
+	var/list/prepared_steps = prepare_step(source)
+	if(isnull(prepared_steps))
+		return
+
+	var/footstep_type = null
+	var/list/footstep_sounds
+	var/stepcount = steps_for_living[source]
+	var/obj/item/bodypart/leg/left_leg = source.get_bodypart(BODY_ZONE_L_LEG)
+	var/obj/item/bodypart/leg/right_leg = source.get_bodypart(BODY_ZONE_R_LEG)
```
Comment (Ghommie) [inline code/datums/elements/footstep.dm:146]: These should be null and only be assigned values once the if statement for body parts covered doesn't pass. This obviously mean adding a layer of indentation but it's a mild opt.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Pull request merged**